### ISSUE TITLE
Pool should not log an ERROR if hkl is not installed

### ISSUE THREAD

**Issue by cpascual** [sardana-org/sardana#394]
If the HKL library is not installed, any invocation of the Pool logs this error (which is otherwise harmless , in the sense that it just indicates that you cannot use HKL stuff):

MainThread     ERROR    2016-12-22 10:32:41,590 ModuleManager: Invalid module 

IMHO, the error message is confusing, though. It should be an INFO or WARNING at most instead  (or even probably just a DEBUG). As it is, the message is likely to mislead in case of unrelated problems.

**Comment by teresanunez** [comment]
I can send a PR for solving this. But only to check if you agree with me, I think it should be a warning, 
so that the people can see it when they start the Pool per hand and they are aware that they can not use the diffractometer. This is also consistent with what happens if one controller has an error (in this case it is not exactly an error, only a missing library that is just needed for that controller), just now one gets something like:
MainThread     ERROR    2016-12-22 12:29:38,177 ModuleManager: Invalid module ElecMeter
MainThread     ERROR    2016-12-22 12:29:38,178 ModuleManager: Invalid module Simu0DCtrl
MainThread     ERROR    2016-12-22 12:29:38,248 ModuleManager: Invalid module IK220_PseudoCountersLib
MainThread     ERROR    2016-12-22 12:29:38,248 ModuleManager: Invalid module LI_BCM_PCCtrl
MainThread     ERROR    2016-12-22 12:29:38,257 ModuleManager: Invalid module MythenController
MainThread     ERROR    2016-12-22 12:29:38,278 ModuleManager: Invalid module SimuIOCtrl
MainThread     ERROR    2016-12-22 12:29:38,279 ModuleManager: Invalid module AttenuatorIOCtrl
MainThread     ERROR    2016-12-22 12:29:38,294 ModuleManager: Invalid module HasyVirtualCounterCtrl
MainThread     ERROR    2016-12-22 12:29:38,319 ModuleManager: Invalid module TimestampCTCtrl
MainThread     ERROR    2016-12-22 12:29:38,338 ModuleManager: Invalid module NI6602CTCtrl
MainThread     ERROR    2016-12-22 12:29:38,339 ModuleManager: Invalid module Vct6Ctrl
MainThread     ERROR    2016-12-22 12:29:38,344 ModuleManager: Invalid module LocumCoTiCtrl
MainThread     ERROR    2016-12-22 12:29:38,347 ModuleManager: Invalid module MythenCoTiCtrl
MainThread     ERROR    2016-12-22 12:29:38,398 ModuleManager: Invalid module SocketCommunicationController
MainThread     ERROR    2016-12-22 12:29:38,399 ModuleManager: Invalid module SerialCommunicationController
MainThread     ERROR    2016-12-22 12:29:38,400 ModuleManager: Invalid module NullCommunicationController

when the Pool is started, what also doesn't mean that you can not run the Pool.
I think if we put it in DEBUG there are so many messages that you will never find it and in info
 is quite 'light' I think, you will not look for something in info if you have a problem for creating a
hkl controller and you don't know the reason.
 Well, just tell me your opinion and then I will proceed with the corresponding changes. Thanks.

**Comment by cpascual** [comment]
Regarding the severity:

I think there is a difference between the other controllers you mention and the HKL one, which is that the HKL one is included in the core library, while the other ones need to be added explicitly. So in a default installation, whithout adding any extra controller paths, the only ERROR/WARNING comes from HKL, which makes it very prominent. 

But still, using "WARNING" is fine for me

**Comment by reszelaz** [comment]
The request makes sense to me - we should not scare the user with the ERROR message knowing that the HKL controller is optional.

In the #409 I see one effect which is not optimal. Yes, we have eliminated the ERROR message in favor of WARNING one, but the controller code is actually loaded to the Pool. Making it possible to the user to create instances of this controller or at least giving him this impression. This is because the interpretation of the module does not give any error now.

Following Carlos' suggestion, we could change the `ControllerManager`/`ModuleManager` to use different severity when logging problems with controllers included in the core and controllers from the additional `PoolPath`. But I'm not 100% convinced yet :). This way we would not need to make any special protection in the HKL controller code and leave the logging to the manager class.

Well, it looked trivial to fix it, but I still have some doubts. What do you think?

**Comment by teresanunez** [comment]
I agree, I also think that it is better to keep the hkl controller like it was, and handle the error message
in the ModuleManager. Should I remove (if possible) the PR.?.

**Comment by teresanunez** [comment]
I will close the PR.
